src/Microsoft.Azure.WebJobs.Host/Exceptions/RecoverableException.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Serialization;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// A recoverable exception, i.e. can be handled.
+    /// </summary>
+    [SuppressMessage("Microsoft.Design", "CA1032:ImplementStandardExceptionConstructors")]
```

Comment:
why this exclusion? it seems like you've implemented everything you need. Note that our other exceptions are following the design guidelines